roachtest: schemachange/mixed/tpcc failed

SHA: https://github.com/cockroachdb/cockroach/commits/51a6fdedf0ce1d1329d40d801a7deaf8206b6b07

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/mixed/tpcc PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1436116&tab=buildLog

```
The test failed on branch=provisional_201908060405_v19.1.4, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190812-1436116/schemachange/mixed/tpcc/run_1
	schemachange.go:452,schemachange.go:414,cluster.go:2078,errgroup.go:57: pq: syntax error at or near "primary"
	cluster.go:1735,tpcc.go:183,cluster.go:2078,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1565651234-57-n5cpu16:5 -- ./workload run tpcc --warehouses=1000 --histograms=perf/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=3h0m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		       0           26.0           28.9    184.5  38654.7  45097.2  45097.2 stockLevel
		  14m27s        0           35.9           28.8   1543.5  12884.9  16106.1  16106.1 delivery
		  14m27s        0          349.6          288.0   3623.9  38654.7  68719.5  73014.4 newOrder
		  14m27s        0           44.8           29.0     83.9   6442.5  40802.2  40802.2 orderStatus
		  14m27s        0          410.3          282.8   6442.5  81604.4 103079.2 103079.2 payment
		  14m27s        0           37.8           28.9    218.1   6442.5  42949.7  42949.7 stockLevel
		  14m28s        0           29.1           28.8   1610.6  16643.0  24696.1  24696.1 delivery
		  14m28s        0          169.3          287.9   3355.4  22548.6  73014.4  98784.2 newOrder
		  14m28s        0           15.0           29.0     46.1   9126.8  18253.6  18253.6 orderStatus
		  14m28s        0          278.5          282.8   9663.7 103079.2 103079.2 103079.2 payment
		  14m28s        0           16.0           28.9    419.4   6442.5  12348.0  12348.0 stockLevel
		: signal: killed
	cluster.go:2099,tpcc.go:193,schemachange.go:410,test_runner.go:691: Goexit() was called

```

SHA: https://github.com/cockroachdb/cockroach/commits/62b1678f652461bbc1aaf6bc2c0dd03105ce0ebe

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/mixed/tpcc PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1488785&tab=buildLog

```
The test failed on branch=40765, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190914-1488785/schemachange/mixed/tpcc/run_1
	cluster.go:1735,tpcc.go:184,cluster.go:2091,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1568493418-85-n5cpu16:5 -- ./workload run tpcc --warehouses=1000 --histograms=perf/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=3h0m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		n.go:447  error in newOrder: dial tcp 10.128.0.118:26257: connect: connection refused
		 9647.0s    21165            1.0           26.6 103079.2 103079.2 103079.2 103079.2 delivery
		 9647.0s    21165            1.0          265.8 103079.2 103079.2 103079.2 103079.2 newOrder
		 9647.0s    21165            1.0           26.6 103079.2 103079.2 103079.2 103079.2 orderStatus
		 9647.0s    21165            0.0          265.4      0.0      0.0      0.0      0.0 payment
		 9647.0s    21165            2.0           26.6 103079.2 103079.2 103079.2 103079.2 stockLevel
		 9648.0s    21165            0.0           26.6      0.0      0.0      0.0      0.0 delivery
		 9648.0s    21165            1.0          265.8 103079.2 103079.2 103079.2 103079.2 newOrder
		 9648.0s    21165            0.0           26.6      0.0      0.0      0.0      0.0 orderStatus
		 9648.0s    21165            0.0          265.3      0.0      0.0      0.0      0.0 payment
		 9648.0s    21165            0.0           26.6      0.0      0.0      0.0      0.0 stockLevel
		: signal: killed

```

SHA: https://github.com/cockroachdb/cockroach/commits/62b1678f652461bbc1aaf6bc2c0dd03105ce0ebe

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/mixed/tpcc PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1489712&tab=buildLog

```
The test failed on branch=40765, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190915-1489712/schemachange/mixed/tpcc/run_1
	cluster.go:1735,tpcc.go:184,cluster.go:2091,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1568575032-78-n5cpu16:5 -- ./workload run tpcc --warehouses=1000 --histograms=perf/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=3h0m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		999          149.0          277.3  26843.5 103079.2 103079.2 103079.2 payment
		  903.0s    11999           13.0           28.3    704.6  18253.6  38654.7  38654.7 stockLevel
		E190916 01:01:13.182063 1 workload/cli/run.go:447  error in payment: ERROR: result is ambiguous (error=rpc error: code = Unavailable desc = transport is closing [propagate]) (SQLSTATE 40003)
		  904.0s    12035           16.0           28.2   1610.6  19327.4  47244.6  47244.6 delivery
		  904.0s    12035          142.0          281.6  12884.9  27917.3  51539.6  51539.6 newOrder
		  904.0s    12035           12.0           28.3    352.3  20401.1  31138.5  31138.5 orderStatus
		  904.0s    12035          173.0          277.1  34359.7 103079.2 103079.2 103079.2 payment
		  904.0s    12035            9.0           28.3    318.8  12348.0  12348.0  12348.0 stockLevel
		E190916 01:01:14.211104 1 workload/cli/run.go:447  error in payment: ERROR: result is ambiguous (error=rpc error: code = Unavailable desc = transport is closing [propagate]) (SQLSTATE 40003)
		: signal: killed

```

--- a/pkg/cmd/roachtest/schemachange.go
+++ b/pkg/cmd/roachtest/schemachange.go
@@ -411,11 +411,18 @@ func makeMixedSchemaChanges(spec clusterSpec, warehouses int, length time.Durati
 				Warehouses: warehouses,
 				Extra:      "--wait=false --tolerate-errors",
 				During: func(ctx context.Context) error {
+					if t.IsBuildVersion(`v19.2.0`) {
+						if err := runAndLogStmts(ctx, t, c, "mixed-schema-changes-19.2", []string{
+							// CREATE TABLE AS with a specified primary key was added in 19.2.
+							`CREATE TABLE tpcc.orderpks (o_w_id, o_d_id, o_id, PRIMARY KEY(o_w_id, o_d_id, o_id)) AS select o_w_id, o_d_id, o_id from tpcc.order;`,
+						}); err != nil {
+							return err
+						}
+					}
 					return runAndLogStmts(ctx, t, c, "mixed-schema-changes", []string{
 						`CREATE INDEX ON tpcc.order (o_carrier_id);`,
 
 						`CREATE TABLE tpcc.customerpks (c_w_id INT, c_d_id INT, c_id INT, FOREIGN KEY (c_w_id, c_d_id, c_id) REFERENCES tpcc.customer (c_w_id, c_d_id, c_id));`,
-						`CREATE TABLE tpcc.orderpks (o_w_id, o_d_id, o_id, PRIMARY KEY(o_w_id, o_d_id, o_id)) AS select o_w_id, o_d_id, o_id from tpcc.order;`,
 
 						`ALTER TABLE tpcc.order ADD COLUMN orderdiscount INT DEFAULT 0;`,
 						`ALTER TABLE tpcc.order ADD CONSTRAINT nodiscount CHECK (orderdiscount = 0);`,

